### PR TITLE
Makeover

### PR DESCRIPTION
**Pull request by safl**
This is primarily:

* An introduction of utilizing **cijoe** via Makefile helpers ``make cijoe-*``
* Change the default behavior of ``make`` to inform the user about the helpers, the build-system etc. instead of "mimicking" the legacy make-behavior of ``./configure && make && make install``.

Additionally, then in preparation to this are a collection of fixes and refactorings reducing the technical debt on the various CIJOE scripts and workflows. What is missing:

- [x] Update the Linux developer tutorial to use the helpers
- [ ] ~~Refactor the GHA to use the make helpers instead of manually invoking cijoe~~
  - Dropped this, since in GHA special output-directories are needed which differ from the default ones used via the Makefile

### CODE REVIEW

File: cijoe/workflows/provision-using-tgz.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- name: xnvme_bindings_py_install_tgz
+  uses: xnvme_bindings_py_install_tgz
+  with:
+    xnvme_source: '{{ config.xnvme.repository.sync.remote_path }}'
+
+# do not build the Python bindings, install those provided via .tgz
+#- name: xnvme_bindings_py_build
+#  uses: xnvme_bindings_py_build
+#  with:
+#    xnvme_source: '{{ config.xnvme.repository.sync.remote_path }}'
```
Comment (karlowich) [inline cijoe/workflows/provision-using-tgz.yaml:67]: If this is not needed, I think we should remove it

**Review by karlowich — CHANGES_REQUESTED** [review]
Overall looking good - a few comments

File: .github/workflows/verify.yml [hunk 2]
```diff
@@ -120,9 +120,18 @@ jobs:
...
+      run: |
+        apt-get update -qy
+        apt-get install -qy curl
+
+    - uses: dtolnay/rust-toolchain@stable
       with:
         toolchain: stable
+        components: clippy, rustfmt
+
+#    - name: Install rust-fmt
```
[the remark this answers is not in the document]
Comment (safl) [reply .github/workflows/verify.yml:133]: Fixed, thanks!

File: .github/workflows/verify.yml [hunk 3]
```diff
@@ -501,14 +507,13 @@ jobs:
...
-        git clone https://github.com/axboe/fio.git
-        cd fio
-        git checkout fio-3.35
-        ./configure
-        gmake -j
+        git clone https://github.com/axboe/fio.git /tmp/fio
+        cd /tmp/fio
+        git checkout fio-3.36
+        git rev-parse --short HEAD
+        gmake
```
[the remark this answers is not in the document]
Comment (safl) [reply .github/workflows/verify.yml:516]: Fixed, thanks!

File: Makefile [hunk 4]
```diff
@@ -126,14 +126,172 @@ docker:
...
+# Create and start a qemu-guest
+endef
+.PHONY: cijoe-guest-setup-blank
+cijoe-guest-setup-blank:
+	@echo "## xNVMe: cijoe-guest-setup-blank"
+	@cd cijoe && cijoe -w workflows/provision-using-git.yaml -l \
+		guest_kill \
+		guest_init \
+		guest_start \
+		guest_check
```
[the remark this answers is not in the document]
Comment (safl) [reply Makefile:182]: I like that it is the same and just using a subset of the steps, then there are fewer workflows to "test".

File: cijoe/configs/debian-bullseye.toml [hunk 5]
```diff
@@ -202,15 +199,16 @@ mount_point = "/mnt/huge"
...
 [fio.repository]
 upstream = "https://github.com/axboe/fio.git"
 path = "/root/git/fio"
-tag = "fio-3.34"
+tag = "fio-3.36"
 
-[fio.build]
-prefix = "/opt/fio"
+#[fio.build]
+#prefix = "/opt/fio"
```
[the remark this answers is not in the document]
Comment (safl) [reply cijoe/configs/debian-bullseye.toml:211]: Hmm, how about adding a comment instead, informing the user how they can add a prefix instead?

File: cijoe/scripts/git_sync.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+Then one needs to transfer the source, or changes to the source, to the remote
+system. This can be done with tools such as rsync and mutagen. However, these
+can cause unwanted behavior when syncing repository data such as ``.git``.
+
+Ideally, one would execute the required git commands, transferring the changes
+to the remote, however, this can be tedious. Thus, this script, running the
+various commands, is doing just that.
+
+Here is what is does:
```
[the remark this answers is not in the document]
Comment (safl) [reply cijoe/scripts/git_sync.py:13]: fixed, thanks.

File: cijoe/configs/debian-bullseye.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (safl) [reply cijoe/configs/debian-bullseye.toml:211]: Went this way...

**Review by mbrsamsung — APPROVED** [review]
Looks very nice. Good job!

**Pull request merged**